public/index.html

```diff
@@ -6,8 +6,9 @@
     <title>The application</title>
     <link rel="stylesheet" href="styles/style.css">
     <script src="scripts/general.js" type="module"></script>
-    <script src="scripts/app.js"  type="module"></script>
-  
+    <script src="scripts/page/mainPage.js" type="module"></script>
```

Comment:
Maybe you should only import the index.html the src="scripts/page/mainPage.js" module, and in mainPage.js import the navbar, and every other functions the main calls.